tempto-core/src/main/java/io/trino/tempto/query/JdbcQueryExecutor.java

```diff
@@ -87,8 +85,13 @@ public QueryResult executeQuery(String sql, QueryParam... params)
     @Override
     public Connection getConnection()
     {
-        if (connection == null) {
-            openConnection();
+        try {
+            if (connection == null || connection.isClosed()) {
```

Comment:
so you can use connection after it closure? and so do you expect that one will close it again?